drivers/include/sdmmc/vendor/sdhc.h

```diff
@@ -0,0 +1,2487 @@
+/**
+ * Copyright (c) 2019 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the Licence at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @brief SD Host Controller Interface Declaration
+ *
+ * @note This file is extracted from the Atmel SAME54 Series Device Support
+ *       Package (1.1.134) http://packs.download.atmel.com/ and slighlty
+ *       modified to be compliant with the SD Host Controller Simplified
+ *       Specification, Version 3.00 [[sdcard.org](https://www.sdcard.org)].
+ *
+ */
+
+#ifndef VENDOR_SDHC_H
+#define VENDOR_SDHC_H
+
+#if !DOXYGEN    /* hide implementation details */
+
+/* SAM0 defines REV_SDHC. In that case following register type are defined by
+ * the Atmel SAM D5x/E5x Series Device Support Package and must not be
+ * redefined */
+#ifndef REV_SDHC

```

Comment:
Although it seems to be SAM0 specific since it is taken from the SAM D5x/E5x Series Device Support Package, it is the base for the SD Host Controller compliant interface definition of the `sdmmc_sdhc` driver, which can be used for any MCU that is compliant with the SD Hoct Controller Specification, such as the EFM32.

Reply:
I didn't want to rewrite the file just to have the interface definition. I felt that using the vendor file under their copyright and modifying it slightly for our purpose was the better option.